lxd/cluster/connect.go

```diff
@@ -64,6 +66,28 @@ func Connect(address string, networkCert *shared.CertInfo, serverCert *shared.Ce
 		args.UserAgent = request.UserAgentNotifier
 	}
 
+	if r != nil {
+		proxy := func(req *http.Request) (*url.URL, error) {
+			ctx := r.Context()
+
+			val, ok := ctx.Value("username").(string)
```

Comment:
constant use here too.

Also I wouldn't be adversed to having a separate set of constants for our HTTP headers in one place so we can see all the ones we use. Although that is probably outside of the scope of this change, and its a lot easier to search the code base for them.